core/trino-main/src/main/java/io/trino/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -565,12 +551,8 @@ public ActualProperties visitExchange(ExchangeNode node, List<ActualProperties>
             Map<Symbol, NullableValue> constants = entries.stream()
                     .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
 
-            ImmutableList.Builder<SortingProperty<Symbol>> localProperties = ImmutableList.builder();
-            if (node.getOrderingScheme().isPresent()) {
-                node.getOrderingScheme().get().getOrderBy().stream()
-                        .map(column -> new SortingProperty<>(column, node.getOrderingScheme().get().getOrdering(column)))
-                        .forEach(localProperties::add);
-            }
+            ImmutableList.Builder<LocalProperty<Symbol>> localProperties = ImmutableList.builder();
```

Comment:
rewrite:
```
List<LocalProperty<Symbol>> localProperties = node.getOrderingScheme()
  .map(OrderingScheme::toLocalProperties);
```

Reply:
In https://github.com/trinodb/trino/pull/7589, I use the builder to add local input properties too